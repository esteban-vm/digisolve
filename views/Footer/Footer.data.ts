import type { NavLinkProps } from './NavLink'
import type { SocialLinkProps } from './SocialLink'

export const navLinks: NavLinkProps[] = [
  {
    id: '97e51480-e825-11ed-b24d-893619994521',
    text: 'About us',
  },
  {
    id: '9c0412a0-e825-11ed-b24d-893619994521',
    text: 'Blog',
  },
  {
    id: 'a1fc3d90-e825-11ed-b24d-893619994521',
    text: 'Help & Support',
  },
  {
    id: 'a4f929e0-e825-11ed-b24d-893619994521',
    text: 'iOS App',
  },
  {
    id: 'a7b0f6e0-e825-11ed-b24d-893619994521',
    text: 'Android App',
  },
]

export const socialLinks: SocialLinkProps[] = [
  {
    id: 'bb690ae0-8c1b-428c-9113-6e22cfb5c44b',
    text: 'Facebook',
    icon: 'facebook-f',
  },
  {
    id: '8083a63a-bd9c-4a89-b7ae-025e873529a8',
    text: 'Twitter',
    icon: 'twitter',
  },
  {
    id: '97bc93f4-ab8c-44dd-a33e-b8c5d6c8260b',
    text: 'TikTok',
    icon: 'tiktok',
  },
  {
    id: '42f20ef7-eabb-485c-8f69-4155f45623e4',
    text: 'Instagram',
    icon: 'instagram',
  },
  {
    id: '14c8210d-8242-48a5-ac7c-be9b5111b213',
    text: 'YouTube',
    icon: 'youtube',
  },
]
