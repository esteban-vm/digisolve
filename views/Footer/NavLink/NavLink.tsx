import type { ComponentWithLink } from '@/types'
import Link from 'next/link'
import { styled } from '@/styles'

type NavLinkProps = {
  text: string
}

const NavLinkComponent: ComponentWithLink<NavLinkProps> = ({ text, url = '#', ...rest }) => {
  return (
    <li {...rest}>
      <Link href={url}>{text}</Link>
    </li>
  )
}

const NavLink = styled(NavLinkComponent)`
  margin-right: 20px;

  a:hover {
    opacity: 0.8;
  }
`

export default NavLink
