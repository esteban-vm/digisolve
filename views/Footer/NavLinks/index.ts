export { default, navLinks } from './NavLinks'
