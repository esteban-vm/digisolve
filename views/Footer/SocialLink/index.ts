export { default /* type SocialLinkProps */ } from './SocialLink'
