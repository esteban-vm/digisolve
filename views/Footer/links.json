{
  "navLinks": [
    { "text": "About us" },
    { "text": "Blog" },
    { "text": "Help & Support" },
    { "text": "iOS App" },
    { "text": "Android App" }
  ],
  "socialLinks": [
    { "icon": ["fab", "facebook-f"], "title": "Facebook" },
    { "icon": ["fab", "twitter"], "title": "Twitter" },
    { "icon": ["fab", "google-plus-g"], "title": "Google Plus" },
    { "icon": ["fab", "instagram"], "title": "Instagram" },
    { "icon": ["fab", "youtube"], "title": "YouTube" }
  ]
}
