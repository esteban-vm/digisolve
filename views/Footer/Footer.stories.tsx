import type { Meta } from '@storybook/react'
import Footer from './Footer'

export default {
  component: Footer,
  title: 'Views/Footer',
} as Meta<typeof Footer>

export const FooterView = () => <Footer />
