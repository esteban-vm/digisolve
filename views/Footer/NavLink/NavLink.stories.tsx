import type { Meta } from '@/stories'
import { Grid } from '@/components'
import { styled } from '@/styles'
import links from '../links.json'
import NavLink from './NavLink'

const Wrapper = styled.div`
  background-color: var(--color-darker);
  padding: 50px;
  font-size: 80%;
  color: var(--color-dark);
  list-style: none;

  ul {
    float: left;

    li {
      display: inline-block;

      a {
        transition: all 500ms;

        :link,
        :visited {
          color: var(--color-dark);
          text-decoration: none;
        }
      }
    }
  }
`

export default {
  component: NavLink,
  title: 'Views/Footer/Nav Links',
  decorators: [
    (Story, context) => {
      return (
        <Wrapper>
          <Grid.Row>
            <Grid.Col isHalf>
              <Story {...context} />
            </Grid.Col>
          </Grid.Row>
        </Wrapper>
      )
    },
  ],
  parameters: {
    viewport: {
      disable: true,
    },
  },
} as Meta<typeof NavLink>

export const Example = () => {
  return (
    <ul>
      {links.navLinks.map((link) => (
        <NavLink key={crypto.randomUUID()} {...link} />
      ))}
    </ul>
  )
}
