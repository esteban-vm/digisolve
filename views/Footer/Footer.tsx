import type { Component } from '@/types'
import { Grid } from '@/components'
import { styled, mediaQuery } from '@/styles'
import { navLinks, socialLinks } from './Footer.data'
import NavLink from './NavLink'
import SocialLink from './SocialLink'

const FooterComponent: Component = (props) => {
  return (
    <footer {...props}>
      <Grid.Row>
        <Grid.Col isHalf>
          <ul className='nav__links'>
            {navLinks.map((link) => (
              <NavLink key={link.id} {...link} />
            ))}
          </ul>
        </Grid.Col>
        <Grid.Col isHalf>
          <ul className='social__links'>
            {socialLinks.map((link) => (
              <SocialLink key={link.id} {...link} />
            ))}
          </ul>
        </Grid.Col>
      </Grid.Row>
      <Grid.Row>
        <p>This webpage has been created for educational purposes.</p>
        <p>You are feel free to use it and customize it to make your own webpages.</p>
      </Grid.Row>
    </footer>
  )
}

const Footer = styled(FooterComponent)`
  background-color: var(--color-darker);
  padding: 50px;
  font-size: 80%;
  color: var(--color-dark);

  ${mediaQuery('md')} {
    padding: 30px;
  }

  ul {
    li {
      display: inline-block;

      a {
        transition: all 500ms;

        :link,
        :visited {
          color: var(--color-dark);
          text-decoration: none;
        }
      }
    }

    &[class$='__links'] {
      ${mediaQuery('xs')} {
        float: none;
        text-align: center;
        margin-bottom: 10px;
      }
    }
  }

  p {
    text-align: center;
    line-height: 1.5;

    ${mediaQuery('md')} {
      line-height: 1.3;
    }

    :first-of-type {
      margin-top: 50px;

      ${mediaQuery('md')} {
        margin-top: 20px;
      }
    }
  }

  .nav__links {
    float: left;
  }

  .social__links {
    float: right;
  }
`

export default Footer
