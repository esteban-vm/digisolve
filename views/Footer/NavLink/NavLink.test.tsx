import { render, cleanup, screen, create } from '@/tests'
import { navLinks } from '../Footer.data'
import NavLink from './NavLink'

describe('ðŸ§ª NAV LINK:', () => {
  const [testLink] = navLinks

  describe('display tests:', () => {
    beforeEach(() => {
      render(<NavLink {...testLink} />)
    })

    afterEach(cleanup)

    it('should be accessible', () => {
      const listItem = screen.getByRole('listitem')
      expect(listItem).toBeInTheDocument()
    })

    it('the text of the link should be displayed', () => {
      const link = screen.getByRole('link', { name: testLink.text })
      expect(link).toBeInTheDocument()
      expect(link).toBeVisible()
    })
  })

  describe('style tests:', () => {
    it('should render with correct styles', () => {
      const tree = create(<NavLink {...testLink} />)
      expect(tree).toMatchSnapshot()
    })
  })
})
