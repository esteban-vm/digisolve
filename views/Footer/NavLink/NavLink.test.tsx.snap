// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª NAV LINK: style tests: should render with correct styles 1`] = `
.emotion-0 {
  margin-right: 20px;
}

.emotion-0 a:hover {
  opacity: 0.8;
}

<li
  className="emotion-0"
  id="97e51480-e825-11ed-b24d-893619994521"
>
  <a
    href="#"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    About us
  </a>
</li>
`;
